core/runtime/v2/manager.go

```diff
@@ -430,10 +431,36 @@ type TaskManager struct {
 }
 
 // NewTaskManager creates a new task manager instance.
-func NewTaskManager(shims *ShimManager) *TaskManager {
-	return &TaskManager{
-		manager: shims,
+func NewTaskManager(ctx context.Context, shims *ShimManager) (*TaskManager, error) {
+	manager := &TaskManager{manager: shims}
+	if err := manager.restoreTasks(ctx); err != nil {
+		return nil, err
+	}
+	return manager, nil
+}
+
+func (m *TaskManager) restoreTasks(ctx context.Context) error {
```

Comment:
is it possible to move this function into `loadShim` process? I was thinking that it's easy to maintain. (it can be handled in follow-up)

Reply:
My hope was to keep shim management (`ShimManager`) separately from task management (`TaskManager`).
`restoreTask` calls shim task streamer.
Moving this to shim manager would require us moving a lot of task specific code to shim manager.

Reply:
sounds good